src/Data/Row/Records.hs

```diff
@@ -408,6 +412,20 @@ uncompose :: forall (f :: * -> *) (g :: * -> *) r . Forall r Unconstrained1
           => Rec (Map (Compose f g) r) -> Rec (Map f (Map g r))
 uncompose = uncompose' @Unconstrained1 @f @g @r
 
+-- | ConstR is used as a type level lambda for coercion.
+newtype ConstR a b = ConstR (Rec a)
+newtype FlipConstR a b = FlipConstR { unFlipConstR :: Rec b }
+
+-- | Coerce a record to a coercible representation.  The 'BiForall' in the context
+-- indicates that the type of every field in @r1@ can be coerced to the type of
+-- the corresponding fields in @r2@.
+coerceRec :: forall r1 r2. BiForall r1 r2 Coercible => Rec r1 -> Rec r2
+coerceRec = unFlipConstR . biMetamorph @_ @_ @r1 @r2 @Coercible @ConstR @FlipConstR @Const Proxy doNil doUncons doCons . ConstR
```

Comment:
Hmm, now that I've proven that this works, I wonder if I should leave the type signature as is but change the implementation to `coerceRec = unsafeCoerce`.  I can leave the old implementation as a comment.